lib/content-types/utils.js

```diff
@@ -25,27 +25,22 @@ const attributes = (config) => {
 
   // map returns either true or a string; filter leaves us with just an array of strings
   const attrs = config.attributes.map(attr => {
-    if ((!attr.hasOwnProperty('name') || attr.name === '')) {
-      let ident = attr.type;
-
-      if (attr.hasOwnProperty('id')) {
-        ident = attr.id;
-      }
-
-      return `Attribute '${ident}' in content type '${config.name}' needs a name`;
-    }
-    if (typeof attr.name !== 'string') {
+    // if there is a name, it must be a string
+    if (attr.hasOwnProperty('name') && attr.name !== '' && typeof attr.name !== 'string') {
       return `Attribute names in content type '${config.name}' must be a string`;
     }
-    if (!attr.id || attr.id === '') {
-      return `Attribute '${attr.name}' in content type '${config.name}' must have an id`;
-    }
-    if (typeof attr.id !== 'string') {
+
+    // if there is an, it must be a string
+    if (attr.hasOwnProperty('id') && attr.id !== '' && typeof attr.id !== 'string') {
```

Comment:
That's only because you've added in new code in this PR that will [generate an `id`](https://github.com/punchcard-cms/content-types/pull/72/files#diff-7632fb88b259681879331dad06f8f231R89) from the `name` of an input plugin, which I'm saying specifically I do not want to do.